libtenzir/include/tenzir/pipeline.hpp

```diff
@@ -538,6 +538,9 @@ class pipeline final : public operator_base {
   /// Returns whether this is a well-formed `void -> void` pipeline.
   auto is_closed() const -> bool;
 
+  /// Splits a pipeline into multiple closed pipelines.
+  auto split_at_void() && -> std::vector<pipeline>;
```

Comment:
@jachris I'm not sure if this'll still be possible with the new exec model. If it isn't, then we should likely introduce some keyword like `---` that we can split at in the AST already.

Reply:
We discussed this in a quick call and agreed that we can always add such a token later if need be. I did make the feature opt-in, though, and removed it from the changelog, as we agreed that this is only desired for tests.